### PR TITLE
Support commenting on table and column in BigQuery

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Support commenting on table and column in BigQuery

## Documentation

(x) Sufficient documentation is included in this PR.

## Release notes

(x) Release notes entries required with the following suggested text:

```markdown
# BigQuery
* Add support for [setting comments on tables](/sql/comment). ({issue}`13882`)
* Add support for [setting comments on columns](/sql/comment). ({issue}`13882`)
```

### CODE REVIEW

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java [hunk 1]
```diff
@@ -502,6 +504,40 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
...
+        BigQueryTableHandle table = (BigQueryTableHandle) tableHandle;
+        BigQueryClient client = bigQueryClientFactory.createBigQueryClient(session);
+
+        RemoteTableName remoteTableName = table.asPlainTable().getRemoteTableName();
+        String sql = format(
+                "ALTER TABLE %s.%s.%s SET OPTIONS (description = %s)",
+                quote(remoteTableName.getProjectId()),
+                quote(remoteTableName.getDatasetName()),
+                quote(remoteTableName.getTableName()),
+                newComment.map(comment -> "'" + comment + "'").orElse("NULL"));
```
Comment (wendigo) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java:519]: what if comment contains ' ?

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java [hunk 2]
```diff
@@ -217,6 +217,19 @@ Job create(JobInfo jobInfo)
         return bigQuery.create(jobInfo);
     }
 
+    public TableResult query(String sql)
+    {
+        log.debug("Execute query: %s", sql);
+        try {
+            // Don't call another query method because setting create disposition in jobs with DDL statements is unsupported
```
Comment (hashhar) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java:224]: Should this comment be on the caller side?

Also should we rename the method to clarify it's purpose? Something like `executeDDL` or `ddlQuery`?

**Review by hashhar — APPROVED** [review]
LGTM % comments

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java:519]: Good catch. I changed to parameterized query and confirmed manually.
```sql
COMMENT ON TABLE test IS 'an''apostrophe';
COMMENT ON COLUMN test.x IS 'an''apostrophe'

SHOW CREATE TABLE test;
              Create Table
-----------------------------------------
 CREATE TABLE bigquery.ebyhr_test.test (
    x bigint COMMENT 'an''apostrophe'
 )
 COMMENT 'an''apostrophe'
(1 row)
```

Filed #14058 for follow-up test cases.

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java:224]: I changed to `void executeUpdate` likes JDBC driver. Also, removed the comment it looks redundant now.

**Review by hashhar — APPROVED** [review]
LGTM.

A question - does BigQuery SDK provide API to set description? I wonder why we use SQL here (which is subject to queueing on the BigQuery side).

**Comment by ebyhr** [comment]
There's no way to set **_only_** description as far as I confirmed. We need to provide whole new table definition to `BigQuery#update`. It may lead to concurrent issue when other job try to modify a table at the same time.

**Pull request merged**